tools/code_format/check_format.py

```diff
@@ -1164,6 +1164,28 @@ def normalize_path(path):
     if format_checker.check_error_messages(ct_error_messages):
         sys.exit(1)
 
+    def check_visibility(error_messages):
+        # https://github.com/envoyproxy/envoy/issues/20589
+        # https://github.com/envoyproxy/envoy/issues/9953
+        # PLEASE DO NOT ADD FILES TO THIS LIST WITHOUT SENIOR MAINTAINER APPROVAL
+        exclude_list = (
+            "-- . ':(exclude)source/extensions/filters/http/buffer/BUILD' "
+            "':(exclude)source/extensions/filters/network/common/BUILD' "
+            "':(exclude)tools/code_format/check_format.py'")
+        command = (
+            "git diff `tools/git/last_github_commit.sh` %s |grep '+.*visibility ='" % exclude_list)
```

Comment:
prefer `$(...)` over ` `...` `

Reply:
re `grep '+.*visibility = '` im wondering if this should/needs to use the `-E` flag - maybe not - ill check now...

Reply:
so re this - i think it does need `-E`

i took a random commit from the log and tested as so...

```console
$ git diff fa0386365f1ee2f9cd4acdad8041d739b5f49a70 -- .  | grep '+.*visibility ='
$ git diff fa0386365f1ee2f9cd4acdad8041d739b5f49a70 -- .  | grep -E '+.*visibility ='
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
     visibility = ["//visibility:public"],
-    visibility = ["//visibility:public"],
-    visibility = ["//visibility:public"],
```

Reply:
do we have buildozer setup? i wonder if we could safely auto-remove all visibility from this dir and see if there's a diff?

Reply:
seems the `-E` is not necessary - i understand now that we are grepping for the patch - and that does appear to work